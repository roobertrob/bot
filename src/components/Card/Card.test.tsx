import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { Card } from '.';

const cardProps = {
  id: 1,
  title: 'doggie',
  running: 0,
  updated_at: '2020-10-29 15:53:45',
  mode: 0,
  simulation: 0,
  stock_codes: 'RAIL3',
  strategy: 'RenkoBot',
  initial_capital: 67692.5,
  number_trades: 29,
  daily_balance: 61.15,
  type: 'Normal',
  created_at: '2020-10-29 15:53:45',
  last_paper: {
    robot_id: 1,
    paper: 'WINBG20',
    position: 5,
    type: 0,
    paper_value: 45.55,
    profit: 0,
    id: 0,
    created_at: '2020-10-29 15:53:45',
  },
  movimentations: [
    {
      date: '2020-10-29 15:53:45',
      value: 62.5,
    },
  ],
};

describe('Card component', () => {
  it('should render card', () => {
    render(<Card {...cardProps} />);
    expect(screen.getByTestId('card-component')).toMatchSnapshot();
    expect(screen.getByTestId('card-component')).toBeVisible();
  });
  it('should render title with text-xs class when length is greater than 28 characters', () => {
    render(
      <Card
        {...cardProps}
        title="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec pharetra sit amet urna id gravida. Fusce nec justo non risus tincidunt porta."
      />,
    );
    expect(screen.getByTestId('header-title')).toHaveClass('text-xs');
  });
  it('should render title with text-base class when length is less than 28 characters', () => {
    render(<Card {...cardProps} />);
    expect(screen.getByTestId('header-title')).toHaveClass('text-base');
  });
   it('should render daily_balance with class text-greenLight', () => {
     render(<Card {...cardProps} />);
     expect(screen.getByTestId('daily-balance')).toHaveClass('text-greenLight');
   });
});
