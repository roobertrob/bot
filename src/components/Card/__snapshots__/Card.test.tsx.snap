// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render card 1`] = `
<div
  class="flex items-center w-96 h-64 bg-white py-3.5 px-5 m-4 shadow-3xl rounded-sm"
  data-testid="card-component"
>
  <div
    class="w-full"
  >
    <header
      class="flex justify-between"
    >
      <span
        class="flex items-center text-body1 font-bold text-base"
        data-testid="header-title"
      >
        doggie
      </span>
      <button
        class="flex items-center space-x-1"
      >
        <span
          class="rounded-full w-2 h-2 bg-orangeWarning"
        />
        <span
          class="text-body3"
        >
          Parado
        </span>
      </button>
    </header>
    <p
      class="text-body4"
    >
      #
      1
    </p>
    <div
      class="text-body4 text-xs flex flex-row space-x-1 mt-[5px]"
    >
      <div
        class="border border-gray2 rounded-sm px-1.5"
      >
        Pessimista
      </div>
      <div
        class="border border-gray2  rounded-sm px-1.5"
      >
        0
         %
      </div>
      <div
        class="border border-gray2  rounded-sm px-1.5"
      >
        RenkoBot
      </div>
    </div>
    <div
      class="flex flex-row border border-gray2 rounded-sm mt-[18px] px-2 items-center justify-between"
    >
      <div
        class="flex flex-row justify-center items-center"
      >
        <div
          class="text-[2rem] text-body2 font-bold ml-4 mr-3.5"
        >
          5
        </div>
        <div
          class="flex flex-col mr-9 py-3.5"
        >
          <div
            class="text-body3 font-bold text-base leading-5"
          >
            WINBG20
          </div>
          <span
            class="text-body1 text-sm leading-5"
          >
            Venda
          </span>
        </div>
      </div>
      <div
        class="flex flex-col"
      >
        <span
          class="text-body4"
        >
          R$ 67.692,50
        </span>
        <span
          class="text-greenLight"
        >
          R$ 45,55
        </span>
      </div>
    </div>
    <footer
      class="mt-6 flex justify-between"
    >
      <div>
        <div
          class="text-body3 text-xs"
        >
          Saldo diário 
        </div>
        <div
          class="font-bold text-greenLight"
          data-testid="daily-balance"
        >
          R$ 61,15
        </div>
      </div>
      <div>
        <div
          class="text-body3 text-xs"
        >
          Trades no dia 
        </div>
        <div
          class="font-bold text-body2 flex justify-end"
        >
          29
        </div>
      </div>
    </footer>
  </div>
</div>
`;
